File: mobile/src/test/java/rhedox/gesahuvertretungsplan/test/localUnitTest/AbbreviationResolverTest.java
package rhedox.gesahuvertretungsplan.test.localUnitTest;

import android.content.Context;
import android.content.res.Resources;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import rhedox.gesahuvertretungsplan.R;
import rhedox.gesahuvertretungsplan.model.AbbreviationResolver;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.when;

/**
 * Created by Robin on 21.05.2016.
 */

@RunWith(MockitoJUnitRunner.class)
public class AbbreviationResolverTest {
    @Mock
    Context context;
    @Mock
    Resources resources;

    @Test
    public void testSubjects() {
        when(context.getApplicationContext()).thenReturn(context);
        when(context.getResources()).thenReturn(resources);
        when(context.getResources().getIdentifier("subjects_ch", "string", null)).thenReturn(R.string.subjects_ch);
        when(context.getResources().getIdentifier("subjects_spo", "string", null)).thenReturn(R.string.subjects_spo);
        when(context.getResources().getIdentifier("subjects_m", "string", null)).thenReturn(R.string.subjects_m);
        when(context.getResources().getIdentifier("subjects_int", "string", null)).thenReturn(R.string.subjects_int);
        when(context.getResources().getIdentifier("subjects_vert", "string", null)).thenReturn(R.string.subjects_vert);
        when(context.getString(R.string.subjects_ch)).thenReturn("Chemie");
        when(context.getString(R.string.subjects_spo)).thenReturn("Sport");
        when(context.getString(R.string.subjects_m)).thenReturn("Mathematik");
        when(context.getString(R.string.subjects_int)).thenReturn("Integration");
        when(context.getString(R.string.subjects_vert)).thenReturn("Vertretungsreserve");

        AbbreviationResolver resolve = new AbbreviationResolver(context);

        assertEquals("Subject 1", "Chemie", resolve.resolveSubject("CH"));
        assertEquals("Subject 2", "Sport", resolve.resolveSubject("sPo"));
        assertEquals("Subject 3", "Mathematik", resolve.resolveSubject("m"));
        assertEquals("Subject 4", "Integration", resolve.resolveSubject("InT"));
        assertEquals("Subject 5", "Vertretungsreserve", resolve.resolveSubject("vert"));
    }

    @Test
    public void testTeachers() {
        when(context.getApplicationContext()).thenReturn(context);
        when(context.getResources()).thenReturn(resources);
        when(context.getResources().getIdentifier("teachers_ro", "string", null)).thenReturn(R.string.teachers_ro);
        when(context.getResources().getIdentifier("teachers_ke", "string", null)).thenReturn(R.string.teachers_ke);
        when(context.getResources().getIdentifier("teachers_xham", "string", null)).thenReturn(R.string.teachers_xham);
        when(context.getResources().getIdentifier("teachers_rue", "string", null)).thenReturn(R.string.teachers_rue);
        when(context.getString(R.string.teachers_ro)).thenReturn("Rosch");
        when(context.getString(R.string.teachers_ke)).thenReturn("Keneder");
        when(context.getString(R.string.teachers_xham)).thenReturn("X-Schinken");
        when(context.getString(R.string.teachers_rue)).thenReturn("Rück");

        AbbreviationResolver resolve = new AbbreviationResolver(context);

        assertEquals("Teacher 1", "Rosch", resolve.resolveTeacher("ro"));
        assertEquals("Teacher 2", "Keneder", resolve.resolveTeacher("kE"));
        assertEquals("Teacher 3", "X-Schinken", resolve.resolveTeacher("xHaM"));
        assertEquals("Teacher 4", "Rück", resolve.resolveTeacher("rü"));
    }
}

File: mobile/src/test/java/rhedox/gesahuvertretungsplan/test/localUnitTest/AbbreviationResolverTest.kt
package rhedox.gesahuvertretungsplan.test.localUnitTest

import android.content.Context
import android.content.res.Resources
import io.mockk.every
import io.mockk.mockk
import org.junit.Assert.assertEquals
import org.junit.Test
import rhedox.gesahuvertretungsplan.R
import rhedox.gesahuvertretungsplan.model.AbbreviationResolver

/**
 * Created by robin on 10.03.2018.
 */
class AbbreviationResolverTest {
    private val context: Context = mockk(relaxed = true)
    private val resources: Resources = mockk(relaxed = true)

    @Test
    fun subjects() {
        every { context.applicationContext } returns context
        every {context.resources} returns resources
        every {context.resources.getIdentifier("subjects_ch", "string", null)} returns R.string.subjects_ch
        every {context.resources.getIdentifier("subjects_spo", "string", null)} returns R.string.subjects_spo
        every {context.resources.getIdentifier("subjects_m", "string", null)} returns R.string.subjects_m
        every {context.resources.getIdentifier("subjects_int", "string", null)} returns R.string.subjects_int
        every {context.resources.getIdentifier("subjects_vert", "string", null)} returns R.string.subjects_vert
        every { context.getString(R.string.subjects_ch)} returns "Chemie"
        every { context.getString(R.string.subjects_spo)} returns "Sport"
        every { context.getString(R.string.subjects_m)} returns "Mathematik"
        every { context.getString(R.string.subjects_int)} returns "Integration"
        every { context.getString(R.string.subjects_vert)} returns "Vertretungsreserve"

        val resolve = AbbreviationResolver(context)

        assertEquals("Subject 1", "Chemie", resolve.resolveSubject("CH"))
        assertEquals("Subject 2", "Sport", resolve.resolveSubject("sPo"))
        assertEquals("Subject 3", "Mathematik", resolve.resolveSubject("m"))
        assertEquals("Subject 4", "Integration", resolve.resolveSubject("InT"))
        assertEquals("Subject 5", "Vertretungsreserve", resolve.resolveSubject("vert"))
    }

    @Test
    fun testTeachers() {
        every {context.applicationContext} returns context
        every {context.resources} returns resources
        every {context.resources.getIdentifier("teachers_ro", "string", null)} returns R.string.teachers_ro
        every {context.resources.getIdentifier("teachers_ke", "string", null)} returns R.string.teachers_ke
        every {context.resources.getIdentifier("teachers_rue", "string", null)} returns R.string.teachers_rue
        every {context.getString(R.string.teachers_ro)} returns "Rosch"
        every {context.getString(R.string.teachers_ke)} returns "Keneder"
        every {context.getString(R.string.teachers_rue)} returns "Rück"

        val resolve = AbbreviationResolver(context)

        assertEquals("Teacher 1", "Rosch", resolve.resolveTeacher("ro"))
        assertEquals("Teacher 2", "Keneder", resolve.resolveTeacher("kE"))
        assertEquals("Teacher 3", "X-Schinken", resolve.resolveTeacher("xHaM"))
        assertEquals("Teacher 4", "Rück", resolve.resolveTeacher("rü"))
    }
}